### PR TITLE
test(typeahead): migrate to TestBed

### PR DESCRIPTION
**Pull request by jnizet**
refs #563

### CODE REVIEW

File: src/typeahead/highlight.spec.ts [hunk 1]
```diff
@@ -28,153 +29,109 @@ function highlightHtml(fixture) {
...
-       const html = '<ngb-highlight result="123" [term]="2" highlightClass="my"></ngb-highlight>';
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('1<span class="my">2</span>3');
-           });
-     })));
+  beforeEach(() => {
+    TestBed.overrideModule(NgbTypeaheadModule, {set: {exports: [NgbHighlight]}});
```
Comment (pkozlowski-opensource) [inline src/typeahead/highlight.spec.ts]: Is this needed?

File: src/typeahead/typeahead.spec.ts [hunk 2]
```diff
@@ -441,7 +384,7 @@ class TestComponent {
   private _objects =
       [{id: 1, value: 'one'}, {id: 10, value: 'one more'}, {id: 2, value: 'two'}, {id: 3, value: 'three'}];
 
-  model = '';
+  model: any = '';
```
Comment (pkozlowski-opensource) [inline src/typeahead/typeahead.spec.ts]: Why this change? Here the type will be inferred as `string`. Please revert.

Comment (jnizet) [inline src/typeahead/typeahead.spec.ts]: One of the tests has

    comp.model = {};

That does not compile anymore if the model is a string.

File: src/typeahead/typeahead.spec.ts [hunk 3]
```diff
@@ -51,360 +51,291 @@ function expectWindowResults(element, expectedResults: string[]) {
...
+         fixture.whenStable()
+             .then(() => {
+               expectInputValue(el, 'text');
 
-             comp.model = null;
-             fixture.detectChanges();
-             fixture.whenStable().then(() => {
+               comp.model = null;
+               fixture.detectChanges();
+               return fixture.whenStable();
```
Comment (pkozlowski-opensource) [inline src/typeahead/typeahead.spec.ts]: This is good stuff :+1:

File: src/typeahead/highlight.spec.ts (hunk 1 above)
Comment (jnizet) [inline src/typeahead/highlight.spec.ts]: Yes, otherwise all the tests fail. My understanding is that the test module is just like any regular module: it imports NgbTypeaheadModule, but NgbHighlight is not exported by NgbTypeaheadModule, so it's an internal component of NgbTypeaheadModule, that may not be used outside of NgbTypeaheadModule, including in the test module. 

Overriding the exports of NgbTypeaheadModule allows exporting the internal NgbHighlight component (i.e. do as if it was exported), and thus using it in the test module.

File: src/typeahead/typeahead.spec.ts (hunk 3 above)
Comment (jnizet) [inline src/typeahead/typeahead.spec.ts]: I had a hard time understanding promise chaining with angular 1. Now is the time to capitalize. I couldn't resist flattening the pyramid of callbacks :-)

**Comment by pkozlowski-opensource** [comment]
Merged, thnx!
